test/test_verify.py

```diff
@@ -304,6 +318,29 @@ def test_verify_succeeds(self, ts_response: TimeStampResponse, verifier: Verifie
             is True
         )
 
+    @pytest.mark.parametrize("ts_response_for_algo", ["sha256", "sha384", "sha512"], indirect=True)
+    def test_verify_message_with_algo(
+        self, ts_response_for_algo: TimeStampResponse, sigstage_certificates
+    ) -> None:
+        builder = VerifierBuilder()
+        for certificate in sigstage_certificates:
+            builder.add_root_certificate(certificate)
```

Comment:
This works, but I'm wondering if it's more proper to call `tsa_certificate()` and `add_intermediate_certificate()`.

Reply:
yeah, I suppose you're right (although in this case its probably `tsa_certificate()` and `add_root_certificate()` to be correct)